### PR TITLE
[Place Page SEO] Add Canonical and Localization HTTP headers

### PR DESCRIPTION
**Pull request by juliawu**
This PR adds:

* A [`rel="canonical"` link](https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls?sjid=16312852956349874621-NC#rel-canonical-header-method) as an HTTP response header. This fix the "duplicate without user selected canonical" errors on the place pages.

* A [`rel="alternate"` link](https://developers.google.com/search/docs/specialty/international/localized-versions#http) to the HTTP response header. This will help the crawler understand we have translated versions of the same page, which will also help reduce "duplicate" errors.

* Tests for both

* Fixes a bug where the summary text is not actually limited to the English locale.

### CODE REVIEW

**Review by dwnoble — COMMENTED** [review]
Nice! Thanks for the thorough testing.

Would adding [x-default](https://developers.google.com/search/docs/specialty/international/localized-versions#xdefault) make sense here too?

Any reason you implemented as headers instead of inline `<link ...>` tags in the html's `<head>` tag?

File: server/routes/place/html.py [hunk 1]
```diff
@@ -99,25 +115,56 @@ def place(place_dcid=None):
...
+    localized_url = CANONICAL_ROOT + flask.url_for('place.place', **
+                                                   canonical_args)
+
+    # Add localized url as a language alternate link to headers
+    link_headers.append(
+        f'<{localized_url}>; rel="alternate"; hreflang="{locale_code}"')
+
+    if locale_code == locale:
+      # Set the url of the current locale as the canonical
+      link_headers.append(f'<{localized_url}>; rel="canonical"')
```
[the remark this answers is not in the document]
Comment (juliawu) [reply server/routes/place/html.py]: Good idea! Extracted to helper function.

**Comment by juliawu** [comment]
> Nice! Thanks for the thorough testing.
> 
> Would adding [x-default](https://developers.google.com/search/docs/specialty/international/localized-versions#xdefault) make sense here too?
> 
> Any reason you implemented as headers instead of inline `<link ...>` tags in the html's `<head>` tag?

I originally left out x-default because the examples made it sound like it was meant for "select a language" pages which we don't have implemented, but upon re-read I think we can point search crawlers to our English page as the default for any non-supported languages. I've added `x-default` now.

Also went with headers instead of inline because I felt the HTML template is a little too messy already, and keeping links in the response header means we don't have to add more bulk to the HTML itself. Google claims to treat headers or inline tags equivalently.

**Review by dwnoble — APPROVED** [review]
Makes sense-thanks Julia!

**Pull request merged**